### PR TITLE
New package: wasmtime-0.19.0

### PR DESCRIPTION
**Pull request by ifreund**

### CODE REVIEW

**Comment by ifreund** [comment]
Enabled only the archs that successfully built on travis. Wish rust was more portable :/

**Comment by ifreund** [comment]
Cleaned up the submodule handling as suggested in #23932

**Comment by ifreund** [comment]
Switched to using `broken=` instead of `arches=` as suggested by @Piraty on IRC

File: srcpkgs/wasmtime/template [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+distfiles="${homepage}/archive/v${version}.tar.gz
+ https://github.com/WebAssembly/wasm-c-api/archive/${_wasm_c_api_commit}.tar.gz
+ https://github.com/WebAssembly/WASI/archive/${_WASI_commit}.tar.gz"
+checksum="f569c74aacd4fe52c5fabe4fda9dd6c41bce72e8e467b52a38626bc0dbad1149
+ aea8cd095e9937f1e14f2c93e026317b197eb2345e7a817fe3932062eb7b792c
+ caf4343207d9f3ec135aae8c113824a8b588db61538d210b22acd6ea7c6d485e"
+
+case ${XBPS_TARGET_MACHINE} in
+	armv7l|armv6l-musl|aarch64-musl)
+		broken="https://travis-ci.org/github/void-linux/void-packages/builds/713040286";;
```
Comment (Piraty) [inline srcpkgs/wasmtime/template:21]: please report this upstream

File: srcpkgs/wasmtime/template [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	x86_64*|i686|aarch64) ;; # known working
```
Comment (Piraty) [inline srcpkgs/wasmtime/template:22]: no need

File: srcpkgs/wasmtime/template [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	*) ;; # status unknown
```
Comment (Piraty) [inline srcpkgs/wasmtime/template:23]: no need as well

File: srcpkgs/wasmtime/template [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+_WASI_commit=2b027d91c29e6141a3768e638076daa23f51621c
+distfiles="${homepage}/archive/v${version}.tar.gz
+ https://github.com/WebAssembly/wasm-c-api/archive/${_wasm_c_api_commit}.tar.gz
+ https://github.com/WebAssembly/WASI/archive/${_WASI_commit}.tar.gz"
+checksum="f569c74aacd4fe52c5fabe4fda9dd6c41bce72e8e467b52a38626bc0dbad1149
+ aea8cd095e9937f1e14f2c93e026317b197eb2345e7a817fe3932062eb7b792c
+ caf4343207d9f3ec135aae8c113824a8b588db61538d210b22acd6ea7c6d485e"
+
+case ${XBPS_TARGET_MACHINE} in
+	armv7l|armv6l-musl|aarch64-musl)
```
Comment (Piraty) [inline srcpkgs/wasmtime/template:20]: i suspect: `armv* | aarch64* ` (and probably ppc as well, but that needs verification)

File: srcpkgs/wasmtime/template (hunk 1 above)
Comment (ifreund) [reply srcpkgs/wasmtime/template:21]: Reported the `aarch64-musl` failure here: https://github.com/bytecodealliance/wasmtime/issues/2133. `armv*` seems to be unsupported.

**Comment by ifreund** [comment]
Set `archs=~armv*` as upstream does not support it judging by this [code](https://github.com/bytecodealliance/wasmtime/blob/0387169934e91722b8b6819773eacc6893718296/crates/runtime/src/traphandlers.rs#L158-L172).

File: srcpkgs/wasmtime/template [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# Template file for 'wasmtime'
+pkgname=wasmtime
+version=0.19.0
+revision=1
+archs="~armv*"
```
Comment (Piraty) [inline srcpkgs/wasmtime/template:5]: judging from [this code](https://github.com/bytecodealliance/wasmtime/blob/0387169934e91722b8b6819773eacc6893718296/crates/runtime/src/traphandlers.rs#L158-L172) you referenced earlier, you want to set `archs="x86_64* i686* aarch64*"`

Comment (ifreund) [reply srcpkgs/wasmtime/template]: Good point, done.

**Comment by ifreund** [comment]
Also changed the `broken=` link to point to the issue opened upstream.

**Pull request merged**